pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1421,7 +1423,7 @@ func (s *TracerSuite) TestUDPPythonReusePort() {
 
 		t.Log(conns)
 
-		return len(conns) == 4
+		require.Len(collect, conns, 4)
```

Comment:
Should these be `assert.*`? The `FailNow` method on `CollectT` exits the program (https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1975), so I am not sure this does what is intended in the PR.

Reply:
since we're providing `collect`, `require.*` will halt the execution of the current iteration, but we will retry after the sleep interval

Reply:
Were you able to test this? Just looking at the code for `CollectT.FailNow` (which gets called from any `require.*` call), I see `runtime.Goexit` being called which would just exit the program (and not continuing after the sleep?).

Reply:
yes, you can try with this example
```go
func TestExample(t *testing.T) {
	i := 0
	require.EventuallyWithT(t, func(ct *assert.CollectT) {
		i++
		t.Log("running iteration", i)
		require.Greater(ct, i, 5)
		require.Equal(ct, 0, i%2)
		t.Log("iteration", i, "done")
	}, time.Second, 100*time.Millisecond)
}
```

The output is
```
=== RUN   TestExample
    tracer_linux_test.go:691: running iteration 1
    tracer_linux_test.go:691: running iteration 2
    tracer_linux_test.go:691: running iteration 3
    tracer_linux_test.go:691: running iteration 4
    tracer_linux_test.go:691: running iteration 5
    tracer_linux_test.go:691: running iteration 6
    tracer_linux_test.go:694: iteration 6 done
--- PASS: TestExample (0.60s)
PASS
```

Reply:
Nvm, `runtime.Goexit` just terminates the current goroutine, so this should be fine.